src/view3d.cpp

```diff
@@ -407,6 +435,69 @@ void View3D::keyPressEvent(QKeyEvent *event)
         // Centre camera on current cursor location
         m_camera.setCenter(m_cursorPos);
     }
+    else if(event->key() == Qt::Key_Z)
+    {
+        // Reset center and zoom to encompass selected geometries
+        QModelIndexList sel = m_selectionModel->selectedRows();
+        if(sel.size() > 0)
+        {
+            Imath::Box3d bbox = m_geometries->get()[sel[0].row()]->boundingBox();
+            for (int i = 1; i < sel.size(); ++i)
+                bbox.extendBy( m_geometries->get()[sel[i].row()]->boundingBox());
+            m_cursorPos       = bbox.center();
+            const double diag = (bbox.max - bbox.min).length();
+            m_animatedViewTransformEndCamera.setCenter(m_cursorPos);
+            m_animatedViewTransformEndCamera.setRotation(m_camera.rotation());
+            m_animatedViewTransformEndCamera.setEyeToCenterDistance(std::max<double>(2*m_camera.clipNear(), diag*0.7));
+            beginAnimateViewTransform();
+        }
+    }
+    else if(event->key() == Qt::Key_A)
+    {
+        // Snap to nearest axis position
+        QQuaternion rot = m_camera.rotation();
+
+        QMatrix4x4 m;
+        QVector3D e;
+        QVector3D newE;
+        double a;
+        QQuaternion rotStep;
+
+        // snap view pos to nearest axis
+        m.setToIdentity();
+        m.rotate(rot);
+        e = QVector3D(0, 0, 1) * m;
+        if(fabs(e.x()) > std::max(fabs(e.y()), fabs(e.z())))
+            newE = QVector3D(Imath::sign(e.x()), 0, 0);
+        else if(fabs(e.y()) > fabs(e.z()))
+            newE = QVector3D(0, Imath::sign(e.y()), 0);
+        else
+            newE = QVector3D(0, 0, Imath::sign(e.z()));
+        e = QVector3D::crossProduct(newE, e);
+        a = asin(e.length());
+        e.normalize();
+        rot *= QQuaternion::fromAxisAndAngle(e, a*180/M_PI); // function wants angle in deg!
+
+        // snap view up to nearest axis
+        m.setToIdentity();
+        m.rotate(rot);
+        e = QVector3D(0, 1, 0) * m;
+        if(fabs(e.x()) > std::max(fabs(e.y()), fabs(e.z())))
+            newE = QVector3D(Imath::sign(e.x()), 0, 0);
+        else if(fabs(e.y()) > fabs(e.z()))
+            newE = QVector3D(0, Imath::sign(e.y()), 0);
+        else
+            newE = QVector3D(0, 0, Imath::sign(e.z()));
+        e = QVector3D::crossProduct(newE, e);
+        a = asin(e.length());
+        e.normalize();
+        rot *= QQuaternion::fromAxisAndAngle(e, a*180/M_PI); // function wants angle in deg!
```

Comment:
This block seems very largely duplicated with the one above.  Let's factor out into a separate helper function?